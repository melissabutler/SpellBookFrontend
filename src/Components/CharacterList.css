.CharacterList {
    text-align: center;
    align-items: center;
    background-color: lightblue;
}

.CharacterList-CreateCharacter {
    background-color: lightblue;
}
.CharacterList-CreateCharacter a {
    color: white;
}

.CharacterList-CreateCharacter a:hover {
    color: black;
}

.CharacterList-CreateCharacter h3 {
    background-color: rgb(41, 77, 144);
    padding: 10px;
    margin: 10px;
    border-radius: 10px;
    
    box-shadow: 3px 3px 3px black;

}
