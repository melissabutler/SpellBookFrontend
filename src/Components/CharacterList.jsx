import React, { useEffect, useContext, useState } from "react";
import CurrentUserContext from "../currentUserContext";
import { Link } from "react-router-dom";


import { Container, Row, Col, Button } from 'react-bootstrap';

import CharacterLink from "./CharacterLink";

import "./CharacterList.css"
import useToggle from "../Hooks/useToggleState";
import CharacterForm from "./CharacterForm";

const CharacterList = ({getUser, createCharacter}) => {
    let currentUser = useContext(CurrentUserContext);
    const [characters, setCharacters] = useState([]);

    const [showForm, toggleForm] = useToggle(false)

    const handleForm = e => {
        e.preventDefault();
        toggleForm();
    }

    useEffect(() => {
        async function getData() {
            let data = await getUser()
            setCharacters([...data.user.characters])
            console.log(data.user)
        }
        getData();
        
    }, []);

    useEffect(() => {
        async function updateList() {
            let data = await getUser();
            setCharacters([...data.user.characters])
            
        }
        updateList()
    }, [showForm])

    return (
        
        <Container className="CharacterList">
            {showForm === false &&
            <Container>
                <Row>
                    <Col>
                        <h1>My Characters</h1>
                    </Col>
                    
                </Row>
                <Row>
                    <Col></Col>
                <Col xs={7}>
                    {characters.map(character => (
                        <Link key={character.id} to={`/characters/${character.id}`}>
                                <CharacterLink key={character.id} character={character}/>
                        </Link>
                    ))}
                
                </Col>
                <Col></Col>
                </Row>
                
                
                <Row className="CharacterList-CreateCharacter">
                    <Col></Col>
                    <Col>
                        <Button onClick={handleForm}>Create a new character!</Button>
                    </Col>
                    <Col></Col>
                </Row>
            </Container> 
            }
            {showForm === true &&
            <CharacterForm toggleForm={toggleForm}  createCharacter={createCharacter}/>
            }
            
        </Container>
    )
}

export default CharacterList;