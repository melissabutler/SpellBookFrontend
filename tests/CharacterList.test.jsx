import React from "react";
import { describe, it, expect, vi } from "vitest";
import { render, screen} from '@testing-library/react'
import CharacterList from "../src/Components/CharacterList";

describe("CharacterList", function() {
        vi.mock('react', async (importOriginal) => {
            const react = await importOriginal('react');
            return {
                ...react,
                useContext: () => ({
                    username: 'u1',
                    email: 'email@email.com',
                    isAdmin: false,
                    characters: [
                        {char_class: "cleric", char_name: "Beebo",id: 1, lvl: 3},
                        {char_class: "cleric", char_name: "Meebo", id: 2,lvl: 3}
                    ]
                })
            
        }})

        vi.mock('react-router-dom', async () => {
            const mod = await vi.importActual('react-router-dom');
            return {
                ...mod, 
                useNavigate: () => "/",
                Link: () => "/"
            }
        })
    it('renders the CharacterList component', async () => {
        render(<CharacterList />)
        screen.debug();
        expect(screen.getByText('My Characters')).toBeInTheDocument();
      
    });


        it('matches snapshot', () => {
            const { asFragment } = render(<CharacterList />);
            expect(asFragment()).toMatchSnapshot();
        })



})
   