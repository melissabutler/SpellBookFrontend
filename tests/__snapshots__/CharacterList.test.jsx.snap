// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CharacterList > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CharacterList container"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h1>
          My Characters
        </h1>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      />
      <div
        class="col-7"
      >
        //
      </div>
      <div
        class="col"
      />
    </div>
    <div
      class="CharacterList-CreateCharacter row"
    >
      <div
        class="col"
      />
      <div
        class="col"
      >
        /
      </div>
      <div
        class="col"
      />
    </div>
  </div>
</DocumentFragment>
`;
